client/src/pages/AdminLogin/index.jsx

```diff
@@ -16,31 +20,37 @@ import {
 import { UPDATE_SNACKBAR } from '../../store/actions/snackbarActions';
 
 import useStyles from './styles';
+import { login, unsuccessfulLogin, setAuth } from './eventHandlers'
 import Copyright from '../../components/Copyright';
 import loginAdmin from './apiCalls';
 
+// eslint-disable-next-line react/prop-types
 const AdminLogin = () => {
   const classes = useStyles();
   const dispatch = useDispatch();
+  const history = useHistory();
+  const location = useLocation();
+
+  const { from } = location.state || { from: { pathname: "/admin" } };
 
   const { helperText, data } = useSelector((state) => state.adminLoginReducer);
   const { username, password } = data;
 
+  useEffect(() => {
+    const { auth } = data;
+    if (auth){
+      setTimeout(() => history.replace(from), 200);
```

Comment:
Why are you using `setTimeout` here?

Reply:
Good spot! I think I was testing for appearances for 'fake async' as currently stuff would load to quickly - if that is a thing. 
Removed!